Automattic/wp-calypso
# Checkout: redirect to content form after BBE purchase

Fixes https://github.com/Automattic/martech/issues/1732

## Proposed Changes

If coming from a signup flow, the user is redirected to the site content form after a BBE purchase is complete. This is automatically handled by the `signup destination` feature which internally uses browser storage. However, if the user exits the checkout page while leaving items in the cart, and then resumes checkout, they are redirected to the thank you page.

In this PR, we add an `afterPurchaseUrl` key to the `extra` item of the cart product. This is used by checkout to perform the redirect after the purchase is complete.

Also, since the thank you page is outdated and not needed anymore, it has been removed in this PR.

## Testing Instructions



* Go to `/start/do-it-for-me`.
* Complete the steps till you reach the checkout page.
* Click on the back arrow, and then select "Leave Items" in the confirmation dialog.
* Click on the cart icon in the Calypso masterbar and open checkout.
* Complete the purchase.
* Confirm that you are redirected to the site content form.

-----

#### Testing for regressions
* Confirm that you are able to complete the purchase and the site content form is shown while using the regular signup flow (`/start/do-it-for-me`) in one go.

## Pre-merge Checklist
